### PR TITLE
Apothecary - C standard 17 and CXX Standard 23 - variables

### PR DESCRIPTION
**Pull request by danoli3**
C standard 17 
CXX Standard 23 

Setup variables and set in apothecary for define default 
Can be overloaded with 
```
if [ -z "${C_STANDARD+x}" ]; then
    export C_STANDARD=17
fi

if [ -z "${CPP_STANDARD+x}" ]; then
    export CPP_STANDARD=2b
fi
```

- [OpenCV [4.9.0]->[4.10]
+ OpenCV Contrib modules
- FreeImage 3.19.5 - ARM64 fixes
- Msys2 fixes for packaged code 
- formulae updates 

- GitHub actions env.USE_ARTIFACT to enable cached version use when building or not - even if no version change
needed as no version changes for many libraries, however CXX change

### CODE REVIEW

**Comment by danoli3** [comment]
FreeImage for MSVC has this bug for C++20 :
```
(ClCompile target) -> 
  D:\a\apothecary\apothecary\apothecary\build\FreeImage\Source\MapIntrospector.h(98,36): error C2065: '_Node': undeclared identifier [D:\a\apothecary\apothecary\apothecary\build\FreeImage\build_vs_64\FreeImage.vcxproj]
  D:\a\apothecary\apothecary\apothecary\build\FreeImage\Source\MapIntrospector.h(98,36): error C3861: '_Node': identifier not found 
  ```

Looks like @elvisdukaj  was on the case already and patched it. Merging at danoli3/FreeImage (patched) repo 
https://github.com/danoli3/FreeImage/pull/7

**Review by danoli3 — COMMENTED** [review]
Looks good. Need to disable some modules in opencv contrib though will report upstream

**Comment by danoli3** [comment]
Reported opencv contrib arm64 bug https://github.com/opencv/opencv_contrib/issues/3767 disabled for now (just rgbd)

**Comment by danoli3** [comment]
[OpenCV [4.9.0]->[4.10]
OpenCV Contrib modules
FreeImage 3.19.5 - ARM64 fixes

Msys2 fixes for packaged code

formulae updates

GitHub actions env.USE_ARTIFACT to enable cached version use when building or not - even if no version change
needed as no version changes for many libraries, however CXX change

File: apothecary/formulas/svgtiny/svgtiny.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 # uses a makeifle build system
 
-FORMULA_TYPES=( "linux64" "linuxarmv6l" "linuxarmv7l" "linuxaarch64" "osx" "vs" "ios" "watchos" "catos" "xros" "tvos" "android" "emscripten" "msys2" )
```
Comment (ofTheo) [inline apothecary/formulas/svgtiny/svgtiny.sh:10]: @danoli3 do you remember why msys2 was removed from the svgtiny formula? 
I don't think there is a msys2 / mingw  package for it. 
cc @dimitre

**Pull request merged**